### PR TITLE
github-bots: optionally use a pre-defined service account email

### PR DESCRIPTION
**Pull request by hectorj2f**
Allow passing a service account instead of always creating a service account.

### CODE REVIEW

File: modules/github-bots/main.tf [hunk 1]
```diff
@@ -9,7 +9,9 @@ module "service" {
   name            = var.name
   project_id      = var.project_id
   regions         = var.regions
-  service_account = google_service_account.sa.email
+
```
Comment (k4leung4) [inline modules/github-bots/main.tf:12]: please change it such that when passing in a service account, have it not create the one here.

File: modules/github-bots/main.tf [hunk 2]
```diff
@@ -1,4 +1,5 @@
 resource "google_service_account" "sa" {
+  count        = var.service_account_email == "" ? 1 : 0
```
Comment (k4leung4) [inline modules/github-bots/main.tf:2]: as this changed the resource path, 
can you add a moved block
https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#enabling-count-or-for_each-for-a-resource

```
moved {
  from = google_service_account.sa
  to = google_service_account.sa[0]
}
```

Comment (hectorj2f) [reply modules/github-bots/main.tf:2]: Oh! nice!

**Comment by rawlingsj** [comment]
I tested this patch against an existing bot + authorize service and the SA wasn't changed so AFAICT this won't affect existing deployments that use the default created SA.

**Comment by rawlingsj** [comment]
@k4leung4 looks ok - you happy to merge?

File: modules/github-bots/variables.tf [hunk 3]
```diff
@@ -123,3 +123,9 @@ variable "deletion_protection" {
   description = "Whether to enable delete protection for the service."
   default     = true
 }
+
+variable "service_account_email" {
+  description = "The email of the service account being authorized to invoke the private Cloud Run service."
```
Comment (imjasonh) [inline modules/github-bots/variables.tf:128]:
```suggestion
  description = "The email of the service account being authorized to invoke the private Cloud Run service. If empty, a service account will be created and used."
```

**Review by k4leung4 — APPROVED** [review]
lets try this
have a feeling this might not work

**Pull request merged**